src/policy/concrete.rs

```diff
@@ -805,6 +870,36 @@ where
     }
 }
 
+/// Average satisfaction cost for [`TapTree`] with the leaf [`Miniscript`] nodes having
+/// probabilities corresponding to the (sub)policies they're compiled from.
+///
+/// Average satisfaction cost for [`TapTree`] over script-spend paths is probability times
+/// the size of control block + the script size.
+#[cfg(feature = "compiler")]
+fn taptree_cost<Pk: MiniscriptKey>(
+    tr: &TapTree<Pk>,
+    ms_cache: &MsTapCache<Pk>,
+    policy_cache: &PolicyTapCache<Pk>,
+    depth: u32,
```

Comment:
Why is there a depth parameter here? calculating the cost of the tree at a depth? But then what does average mean?

Reply:
Looks like this is always called with value 0. We should remove it. Perhaps you need a helper around it.

Reply:
Addressed in d384b63